### ISSUE TITLE
docker 13 breaks docker::compose, compose uses static version and requires an internet connection...

### ISSUE THREAD

**Issue by Shoeler** [garethr/garethr-docker#627]
```puppet
class {'docker::compose':
    ensure       => present,
    install_path => '/bin',
}
```

I had to hack docker::compose in the meantime to be a file resource, but that's gross on my part because I have to update docker-compose in my puppet repo every time they change docker.

Additionally, the module currently defaults to using 1.7.0 in params.pp without any logic as to the installed version of docker.  I had to override that with some logic.  It would be better if that pulled the latest version somehow.

**Comment by Shoeler** [comment]
I found another fun bit - docker::compose seems current broken even with the new docker-compose.  I'm guessing they changed a docker command it used.  My resource:

```
  docker_compose { $composefile:
    ensure    => present,
    require   => File[$composefile],
    subscribe => File[$composefile],
  }
```

Info: Checking for compose project data
Debug: Executing: '/usr/bin/docker ps --format {{.Label "com.docker.compose.service"}} --filter label=com.docker.compose.project=data'
Error: /Stage[main]/Solr::Docker/Docker_compose[/data/docker-compose.yml]: Could not evaluate: Execution of '/usr/bin/docker ps --format {{.Label "com.docker.compose.service"}} --filter label=com.docker.compose.project=data' returned 1: template: :1:2: executing "" at <.Label>: Label is not a method but has arguments

**Comment by Shoeler** [comment]
Hey yet more information - looks like the broken docker::compose is a bug:

https://github.com/docker/docker/issues/28337

**Comment by BarnumD** [comment]
Specify the docker 12 version in your role as a workaround: 
  class {'docker':
    version => '1.12.6-0~ubuntu-xenial',
  }

**Comment by BarnumD** [comment]
Fixed for me with docker 1.13.1.
